File: leet_code/math/507/507.dart
import 'dart:math';

class Solution {
  bool checkPerfectNumber(int num) {
    if (num <= 1) {
      return false;
    }
    
    int divisorsSum = 1; 
    

    for (int i = 2; i <= sqrt(num); i++) {
      if (num % i == 0) {
        divisorsSum += i;
        if (i != num ~/ i) {  
          divisorsSum += num ~/ i;
        }
      }
    }
    
    return divisorsSum == num;
  }
}

File: leet_code/math/507/507.py
import math

class Solution:
    def checkPerfectNumber(self, n: int) -> bool:
        if n <= 1:
            return False
        divisors_sum = 1  
        for i in range(2, int(math.sqrt(n)) + 1):
            if n % i == 0:
                divisors_sum += i
                if i != n // i:  
                    divisors_sum += n // i
        
        return divisors_sum == n
